ci/util/serial_probe.py

```
@@ -0,0 +1,293 @@
+    (0x303A, 0x1001): "Espressif native USB CDC (ESP32-S3/C3/C6/H2/P4)",
+    # CP2102 from Silicon Labs — common on ESP32-WROOM dev kits.
+    (0x10C4, 0xEA60): "Silicon Labs CP2102 USB-UART (ESP32 dev kit)",
+    # CH340/CH341 from WCH — common on cheaper ESP32/Arduino clones.
+    (0x1A86, 0x7523): "WCH CH340 USB-Serial",
+    (0x1A86, 0x55D4): "WCH CH343 USB-Serial",
+    # FTDI FT232R — older Arduinos, some shields.
+    (0x0403, 0x6001): "FTDI FT232R USB-UART",
+    (0x0403, 0x6010): "FTDI FT2232 dual UART",
+    (0x0403, 0x6014): "FTDI FT232H USB-UART",
+    # Teensy 3.x/4.x.
+    (0x16C0, 0x0483): "PJRC / Teensy USB-Serial (also LPC11U35 VCOM)",
```

Ruff (0.15.17)

[error] Dictionary key literal `(0x16C0, 0x0483)` repeated

(F601)